lib/db/mysql/index.js

```diff
@@ -27,6 +33,30 @@ function MysqlStore(options) {
     return next();
   };
   this._pool = mysql.createPool(options);
+
+  // Enforce sane defaults on every new connection.
+  // These *should* be set by the database by default, but it's nice
+  // to have an additional layer of protection here.
+  this._pool.on('connection', function(connection) {
```

Comment:
In theory the overhead of doing this won't be too bad, because we only do it when we create a new connection for the pool, and we then re-use that connection for many queries.

One downside of doing this by listening to an event though, is that we can't block until the query has been completed.  So there'd probably be a race condition here, where calling code could use the connection in between the `SELECT @@sql_mode` and the `SET SESSION sql_mode`.